uk/#3561-skimsUsageStats

Ummul-Khair opened the pull request:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned

Currently, collection is done for ODSkims, ParkingSkims, FreightSkims, TAZSkims, DriveTimeSkims, TransitCrowdingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3607)
<!-- Reviewable:end -->

Ummul-Khair committed:
Corrected issue of multiple displays of warnings. Warnings are displayed on 0th iteration.

Ummul-Khair committed:
Merge remote-tracking branch 'origin/develop' into develop

Ummul-Khair committed:
Merge remote-tracking branch 'origin/develop' into develop

Ummul-Khair committed:
Merge remote-tracking branch 'origin/develop' into develop

Ummul-Khair committed:
Merge remote-tracking branch 'origin/develop' into develop

Ummul-Khair committed:
Merge remote-tracking branch 'origin/develop' into develop

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

nikolayilyin commented on src/main/scala/beam/router/skim/core/AbstractSkimmer.scala:
```diff
@@ -65,10 +65,21 @@ abstract class AbstractSkimmerReadOnly extends LazyLogging {
   private[core] val pastSkimsInternal = mutable.HashMap.empty[Int, Map[AbstractSkimmerKey, AbstractSkimmerInternal]]
-
+  var numberOfRequests: Int = 0
+  var noOfReturnsSkimMapActualValue: Int = 0
```
I would call it like numberOfSkimValueFound, or similar

Ummul-Khair replied:
Alright, thanks. I'm also not sure about the code written for generation of activities. I hope that is what you were hoping for. When I ran it for the beamville and sf-light scenarios I noticed that only OD skims had requests. The values for other skims were 0.

nikolayilyin replied:
The low usage of the rest of the skims is expected.
In order to turn generation of secondary activities ON you need to set this flag to true
beam.agentsim.agents.tripBehaviors.multinomialLogit.generate_secondary_activities

In order to get some skims usage during generation of activities you will need to do a warm start simulation.

nikolayilyin reviewed (commented):
Looks good so far

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair commented:
test!

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair commented:
test!

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair commented:
test!

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair commented:
test!

Ummul-Khair commented:
test!

nikolayilyin commented on src/main/scala/beam/router/skim/core/AbstractSkimmer.scala:
```diff
@@ -65,10 +65,24 @@ abstract class AbstractSkimmerReadOnly extends LazyLogging {
-
+  var numberOfRequests: Long = 0
+  var numberOfSkimValueFound: Long = 0
+  var numberOfDefaultSkimValueFound: Long = 0
```
I do not see where 'numberOfDefaultSkimValueFound' is calculated, it seems it will always be 0.
Giving it a second thought - let's remove the `numberOfDefaultSkimValueFound` statistic at all, two of the rest is enough.

nikolayilyin commented on src/main/scala/beam/router/skim/readonly/TransitCrowdingSkims.scala:
```diff
@@ -87,7 +87,7 @@ class TransitCrowdingSkims(vehicleTypes: Map[Id[BeamVehicleType], BeamVehicleTyp
       x.get(key).asInstanceOf[Option[TransitCrowdingSkimmerInternal]]
     }
-
+    numberOfRequests = numberOfRequests + 1
```
Let's actually calculate the rest of statistics for this skimmer too, let's assume each time either a value in aggregatedFromPastSkims or in the pastSkims found we increase numberOfSkimValueFound, otherwise let's assume it is a default value returned

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

Ummul-Khair committed:
Collected usage statistics of skims:
1. number of requests
2. number of times the actual value from skims map was returned
Currently, collection is done for ODSkims, ParkingSkims and RideHailSkims

Resets stats at the beginning of each iteration, writes stats to logs at the end of each iteration. Writes stats to logs after generation of activities

nikolayilyin reviewed (approved):
lgtm

nikolayilyin commented:
test!

nikolayilyin commented:
test!
